rust/arrow/src/array/array.rs

```diff
@@ -622,6 +621,51 @@ def_numeric_from_vec!(
     DataType::Time64(TimeUnit::Nanosecond)
 );
 
+impl<T: ArrowTimestampType> PrimitiveArray<T> {
+    pub fn from_vec(data: Vec<i64>, timezone: Option<Arc<String>>) -> Self {
```

Comment:
According to arrow's specification, only timestamp has timezone field, so maybe this only applies to TimestampArray. Or at least we should add some doc to state that timezone only applies to Timestamp array, others will be ignore. Also this implementation has much duplication with `From<Vec<i64>>`, so I think we should replace `From<Vec<i64>>`'s implementation with this one.

Reply:
The `ArrowTimestampType` trait covers that as I've made it to only apply to timestamp types. So the `from_vec` is only available on Timestamp arrays.
I've removed the `From<Vec<..>>`s in favour of these ones.